# Problem 1

##A
We know the defitintion of redshift Z is 
Z = (λobserved - λemitted)/( λobserved) 

Then from this let us define define Z2, the redshift observed for the Galaxy 2 as
Z2  = ( λ1,o - λ2,e)/( λ1,o) 
where 
λ1,o is the wavelength observed by Galaxy 1
λ2,e is the wavelength emitted by Galaxy 2

and Z1  = ( λ0,o - λ1,e)/( λ0,o) 
where
λ0,o is the wavelength observed by the observer
λ1,e is the light emitted by Galaxy 1

Rearranging Z into the form 1-Z = λe/λo we get 
Z1 = 1 - (λ1,e/λ0,o)  and Z2 = 1 - (λ2,e/λ1,o)

Then we an rearrange them inorder to get 
λ1,e = λ0,o(1-Z1) {1}
and 
λ2,e = (1-Z2)λ1,o {2}

Then comes a very important relation. 
λ1,o = λ1,e 
in other words, the amount of light which Galaxy 1 observes, is that same amout of light which is emitted by it from the perspective of the observer. Thus we can plug {1} into {2} and get; 
λ2,e = (1-Z2) * λ0,o(1-Z1) moving variables over we get
λ2,e/λ0,o = (1-Z2) * (1-Z1) 

Expanding out (1-Z2) * (1-Z1) we get
λ2,e/λ0,o = 1 -Z1 - Z2 - Z1Z2 
λ2,e/λ0,o - 1 - Z1 = Z2(-1 - Z1) 

For simplicity, define q = λ2,e/λ0,o 
1 - q = Z2,0 the redshift of Galaxy 2 from the perspective of the observer, usefull for problem B
then we can finnaly get 

Z2 = (Z1 - 1 - q)/(1 + Z1) 



##B 

Start with the definition of anuglar diameter distance
da  = dl /(1+z) 
where dl is the luminosity distance 
dl = (L/4πf)^1/2 = (1+z)Sk(r) 
Thus, 
da = Sk(r)/(1+z) 

Our universe for B is a Einstein-deSitter one, therefore Ωm = 1 and the universe is completly flat.
Therefore Sk = r

da= r/(1+z)

In this case, r is the comoving distance. 
I will now use my code from the previous homework set to solve comoving distnace in a given universe with inputs of redshift


First, I would like to define some terms. 
Z0,1 = 0.5 the redshift of Galaxy 1 from observer
Z0,2 = 1.6 the redshift of Galaxy 2 from the observer

Then using the previous relation 
Z2 = (Z1 - 1 - q)/(1 + Z1) 

1 - q = Z2,0 => q = 1 - Z2,0 
We can solve for the redshift of Z2 from Z1, with our naming convetion it becomes
Z2,1 = (Z1 - 1 - (1 - Z2,0)/(1 + Z1)
Z2,1 = (0.5 - 1 - 1 + 1.6)/(1 + 0.5)
where Z2,1 is the redshift of Galaxy 2 from Galaxy 1

In [17]:
Z01 = 0.5
Z02 = 1.6
Z12 = (0.5 - 1 - 1 + 1.6)/(1 + 0.5) 
print (Z12)

0.06666666666666672


Z12 = 0.66666666672 = 1/15

In [18]:
from scipy.integrate import quad
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

For proper distance we have the integral r = Integral[(cdt)/a] = Integral{0 to z}[(cdz)/H(z)]

We are given that (H(z)/H0)^2 = Ω{r}(1+z)^4 + Ω{m}(1+z)^3 + Ω{lambda} + (1-Ω{o}(1+z)^2

We know that the universe is flat for this problem, so the Ω for curviture goes to 0.

Solving for H(z) we get H(z) = H0*sqrt[Ω{m}(1+z)^3 ]

Plugging into the integral for comoving distance we get

r = integral{0 to z}[(cdz)/(H0sqrt[Ω{m}(1+z)^3])]

We are given that Ω{mass} = 1 for this problem (Einstein-deSitter universe)

In [56]:
H0= 2.2683e-18 #seconds
H0gyr= 0.07154 #gigayears
c = 30000000 #m*s^-1


def integrand(z):
    return (c/(H0*(1*(1+z)**3)**(1/2)))

integral01, err = quad(integrand, 0, Z01)
integral02, err = quad(integrand, 0, Z02)
integral12, err = quad(integrand, 0, Z12)

####*3.24e-23 convert meters to Mpc
r01 = integral01*3.24e-23  #comoving distance to Galaxy 1 from observer
r02 = integral02*3.24e-23   #comoving distnace to Galaxy 2 from observer
r12 = integral12*3.24e-23  #comoving distance to Galaxy 2 from Galaxy 1

print(r01,r02,r12)

157.26784229444988 325.52236467526075 27.214254614996005


then using the equation for angular diameter distance da= r/(1+z) we can get

In [57]:
da01 = r01/(1 + Z01)
da02 = r02/(1 + Z02)
da12 = r12/(1 + Z12)

print(da01, da02, da12)

104.84522819629991 125.2009094904849 25.513363701558756


da01 = 104.845 Mpc

da02 = 125.2009

da12 = 25.5134 Mpc/gigyear

##C
An observation we can make from these calculations is that 
da12 != da02 - da01. 
The simplist way to solve A would to be setting Z12 = Z02-Z01, but comparing to our work so far, we can state this would result in a miscalculation. 
The values are close, as da02 - da01 = 20.6529 Mpc, this is roughly 4/5 of what we measured for da12. This is a surprise, but not unexpected when dealing with the cosmology of galaxies on Mpc distances. Galaxy 1 and Galaxy 2 are closer to each other than the observer is to either of them, thus it is not surprising that they should observe each other closer than we observe them to be. This is due to GR.

## Problem 2

We are given that 
N = 1000 galaxies inside the half mass radius
rh = 1.5 Mpc , the half mass radius
Σ = 10e-3 Mpc^2 

first we find the mean number density of the half mass radius
the volume is given from
4/3 *π * r^2 

from this we arrive that V = 14.137 Mpc^3 (volume of the half mass radius)
then we do n = N/V and arrive at n = 70.735 1/Mpc^3

a simplified equation to calculate mean free path is 
MFP = 1/(Σ * n) = 1/(10e-3 Mpc^2 * 70.735 Mpc^-3) =  14.137 Mpc
This is the average distnace a galaxy travels before it collides with another.

To calculate the time for this to happen we devide by σ, the velocity dispersion of the Coma cluster. σ = 880 km-s^-1.
Since we are dealing with Mpc, we can convert σ to 2.8519e-17 Mpc-s^-1 

thus, we get MFP/σ = 14.137 Mpc/2.8519e-17 Mps-s^-1
the result is *4.957e17s* which we can convert to
*15.7 billions years*

This is greater than than the Hubble time 

The collision time between M31 and the Milky way is as follows
the distance between the two is 700 kpc, and M31 is moving towards the Milky Way at 120km-s^-1. 

Thus to calculate the time of collsion we simply divide the distance by the velocity. 

to make things easier we convert 120km-s^-1 to 3.8889e-15 Kpc/s

doing the math 700Kpc/3.8889e-15 Kpc-s^-1 = 1.7999e17 seconds
which we can convert to *5.7077 billions years*

The collision time between Galaxies in the Coma Cluster is much greater than the time between M31 and the Milky Way, in fact the times between collsions of Galaxies in the Coma Cluster is greater than the hubble time. 

The time between the collision of M31 and the Milky Way is roughly 1/3 of that of the mean collsion time of galaxies in the Comma Cluster